### PR TITLE
feat(issue-platform): Run post_process_group for generic events

### LINKED ISSUE #42530: Add a way to process generic events in `post_process_group`
We need to be able to handle platform issues in `post_process_group`. I think that all platform issues will be processed in the same way, so there's no need to add a new entry for every new `GroupCategory`.

There are two issues here:
 - We need be able to build the `PostProcessJob` for a generic event. Platform issues will only have one `group_id`, same as errors.
 - We need to add a pipeline that will handle all future `GroupCategory` types. Probably we should just have the generic pipeline be a fallback if an explicit pipeline isn't in place. That way we don't need to constantly add new entries into `GROUP_CATEGORY_POST_PROCESS_PIPELINE`, we can just do it if customization is needed for a given category.

### PR DESCRIPTION
**Pull request by rjo100**
Adds generic pipeline to `post_process_group`

Closes https://github.com/getsentry/sentry/issues/42530

### CODE REVIEW

**Comment by rjo100** [comment]
How do I fix the codecov failure?

**Comment by wedamija** [comment]
> How do I fix the codecov failure?

No need, it's not required

**Review by wedamija — APPROVED** [review]

**Comment by rjo100** [comment]
Blocked by test issues. Will update later.

File: tests/sentry/tasks/test_post_process.py [hunk 1]
```diff
@@ -1280,3 +1282,35 @@ def test_process_transaction_event_clusterer(
...
+class PostProcessGroupGenericTest(
+    TestCase,
+    SnubaTestCase,
+    OccurrenceTestMixin,
+    CorePostProcessGroupTestMixin,
+    InboxTestMixin,
+    RuleProcessorTestMixin,
+    SnoozeTestMixin,
+):
+    def create_event(self, data, project_id):
```
Comment (ceorourke) [inline tests/sentry/tasks/test_post_process.py:1297]: Not a blocker but Gilbert made a test util you can use here: https://github.com/getsentry/sentry/blob/master/tests/sentry/issues/test_utils.py#L64

**Pull request merged**